src/averageAges.js

```diff
@@ -36,8 +45,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people.filter(person => withChildren
+    ? (person.sex === 'f'
+      && isMother(people, person))
+    : person.sex === 'f');
+
+  return getAverageAge(women);
+}
+
+function isMother(people, person) {
+  return people.find(child => child.mother === person.name);
 }
```

Comment:
you should create separate functions only in order to not repeat yourself.
In this case you use this logic only once, so there's no need to create a separate function for that

Reply:
OK!! Let it be so!